src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidAccountManagerBroker.cs

```diff
@@ -0,0 +1,455 @@
+        {
+            return _brokerHelper.IsBrokerInstalledAndInvokable();
+        }
+
+        public async Task<MsalTokenResponse> AcquireTokenInteractiveAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenInteractiveParameters acquireTokenInteractiveParameters)
+        {
+
+            CheckPowerOptimizationStatus();
+
+            AndroidBrokerStaticHelper.InteractiveBrokerTokenResponse = null;
+
+            BrokerRequest brokerRequest = BrokerRequest.FromInteractiveParameters(
+                authenticationRequestParameters, acquireTokenInteractiveParameters);
+
+            // There can only be 1 broker request at a time so keep track of the correlation id
+            s_correlationId = brokerRequest.CorrelationId;
+
+            try
+            {
+                await InitiateBrokerHandshakeAsync().ConfigureAwait(false);
+                await AcquireTokenInteractiveViaBrokerAsync(brokerRequest).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                _logger.ErrorPiiWithPrefix(ex, "Android broker interactive invocation failed. ");
+                _brokerHelper.HandleBrokerOperationError(ex);
+            }
+
+            using (_logger.LogBlockDuration("Waiting for Android broker response. "))
+            {
+                await AndroidBrokerStaticHelper.ReadyForResponse.WaitAsync().ConfigureAwait(false);
+                return AndroidBrokerStaticHelper.InteractiveBrokerTokenResponse;
+            }
+        }
+
+        public async Task<MsalTokenResponse> AcquireTokenSilentAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenSilentParameters acquireTokenSilentParameters)
+        {
+            CheckPowerOptimizationStatus();
+
+            BrokerRequest brokerRequest = BrokerRequest.FromSilentParameters(
+                authenticationRequestParameters, acquireTokenSilentParameters);
+
+            try
+            {
+                await InitiateBrokerHandshakeAsync().ConfigureAwait(false);
+                var androidBrokerTokenResponse = await AcquireTokenSilentViaBrokerAsync(brokerRequest).ConfigureAwait(false);
+                return androidBrokerTokenResponse;
+            }
+            catch (Exception ex)
+            {
+                _logger.ErrorPiiWithPrefix(ex, "Android broker silent invocation failed. ");
+                _brokerHelper.HandleBrokerOperationError(ex);
+                throw;
+            }
+        }
+
+        private async Task AcquireTokenInteractiveViaBrokerAsync(BrokerRequest brokerRequest)
+        {
+            using (_logger.LogMethodDuration())
+            {
+                // onActivityResult will receive the response for this activity.
+                // Launching this activity will switch to the broker app.
+
+                _logger.Verbose("Starting Android Broker interactive authentication. ");
```

Comment:
like how neha did [Region discovery] or something at the start of the log messages for regional, i think you did have something similar for android, but might be useful on these log messages to have a short title like [Android Broker]...idk.